crates/ty_ide/src/symbols.rs

```diff
@@ -684,6 +1157,651 @@ if TYPE_CHECKING:
+            .source(
+                "test.py",
+                "from foo import *
+                 TRICKSY = 1
+                 __all__.append('TRICKSY')",
+            )
+            .build();
+        insta::assert_snapshot!(
+            test.exports_for("test.py"),
+            @r"
+        __all__ :: Variable
+        TRICKSY :: Constant
+        ",
+        );
+    }
+
+    /// Tests that a `from module import *` doesn't bring an
+    /// `__all__` into scope if `module` doesn't provide an
+    /// `__all__` that includes `__all__` AND this causes
+    /// `__all__.append` to fail in the importing module
+    /// (because it isn't defined).
+    #[test]
+    fn wildcard_reexport_all_empty_then_added_to_incorrect() {
+        let test = PublicTestBuilder::default()
+            .source(
+                "foo.py",
+                "
+                ZQZQZQ = 1
+                __all__ = []
+            ",
+            )
+            .source(
+                "test.py",
+                "from foo import *
+                 from collections import defaultdict
+                 __all__.append('defaultdict')",
+            )
+            .build();
+        insta::assert_snapshot!(
+            test.exports_for("test.py"),
+            @"",
+        );
+    }
+
+    #[test]
+    fn wildcard_reexport_all_include_all_then_added_to_correct() {
+        let test = PublicTestBuilder::default()
+            .source(
+                "foo.py",
+                "
+                ZQZQZQ = 1
+                __all__ = ['__all__']
+            ",
+            )
+            .source(
+                "test.py",
+                "from foo import *
+                 from collections import defaultdict
+                 __all__.append('defaultdict')",
+            )
+            .build();
+        insta::assert_snapshot!(
+            test.exports_for("test.py"),
+            @r"
+        __all__ :: Variable
+        defaultdict :: Variable
+        ",
+        );
+    }
+
+    #[test]
+    fn wildcard_reexport_all_non_empty_but_non_existent() {
+        let test = PublicTestBuilder::default()
+            .source(
+                "foo.py",
+                "
+                ZQZQZQ = 1
+                __all__ = ['TRICKSY']
+            ",
+            )
+            .source("test.py", "from foo import *")
+            .build();
+        insta::assert_snapshot!(
+            test.exports_for("test.py"),
+            @"",
+        );
+    }
+
+    #[test]
+    fn wildcard_reexport_all_include_all_and_non_existent() {
+        let test = PublicTestBuilder::default()
+            .source(
+                "foo.py",
+                "
+                ZQZQZQ = 1
+                __all__ = ['__all__', 'TRICKSY']
+            ",
+            )
+            .source("test.py", "from foo import *")
+            .build();
+        insta::assert_snapshot!(
+            test.exports_for("test.py"),
+            @"__all__ :: Variable",
+        );
+    }
```

Comment:
(note that I believe this actually causes the `*` import to fail at runtime with `AttributeError`, because it tries to grab `TRICKSY` from `foo` but fails!)

Reply:
Ah yup. I added a comment noting this (and to a couple other tests where this is true too).

Interestingly, pylance behaves the same as ty (now does with this PR): an import is still offered, but it results in a runtime error.

I guess ideally we wouldn't offer completions for such things, but I opened an issue tracking that https://github.com/astral-sh/ty/issues/1760